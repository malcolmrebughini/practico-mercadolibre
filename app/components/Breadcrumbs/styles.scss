@import 'app/styles/vars';


.container {
  color: #999;
  padding: 16px 0;
  margin: 0 auto;

  @media (min-width: 768px) {
    max-width: $body-width;
  }

  & > span:last-of-type {
    font-weight: bold;
  }
}