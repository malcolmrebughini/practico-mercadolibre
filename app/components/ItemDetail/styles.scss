@import 'app/styles/vars';


.container {
  background-color: #fff;
  margin: 0 auto;

  @media (min-width: 768px) {
    max-width: $body-width;
    border-radius: 4px;
  }

  .pic {
    & > img {
      width: 100%;
      height: auto;
    }

    @media (min-width: 768px) {
      width: 680px;
    }
  }

  .condition {
    margin-top: 32px;
    font-weight: 300;
    font-size: 14px;
    text-transform: capitalize;
  }

  .title {
    font-weight: bold;
    font-size: 24px;
    margin-top: 16px;
  }

  .infoContainer {
    padding: 0 32px;

    @media (min-width: 768px) {
      padding: 0;
      flex-shrink: 5;
      padding-left: 32px;
    }
  }

  .descriptionContainer {
    padding: 40px 32px;

    @media (min-width: 768px) {
      padding: 0;
      margin-top: 40px;
    }
  }

  .price {
    font-size: 46px;
    margin: 32px 0;

    .priceDecimals {
      font-size: 23px;
      vertical-align: top;
      line-height: 1.5;
    }
  }

  .buyButton {
    width: 100%;
    background: #3483fa;
    border: 1px solid #3483fa;
    font-size: 18px;
    color: #fff;
    border-radius: 4px;
    padding: 12px;
  }

  .descriptionTitle {
    font-size: 28px;
  }

  .description {
    color: #666;
    margin-top: 32px;
    font-size: 16px;

    @media (min-width: 768px) {
      width: 65%;
    }
  }

  .innerContainer {
    @media (min-width: 768px) {
      display: flex;
    }
  }

  .someContainer {
    @media (min-width: 768px) {
      padding: 32px;
    }
  }
}