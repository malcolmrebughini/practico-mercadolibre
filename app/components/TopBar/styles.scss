@import 'app/styles/colors';
@import 'app/styles/vars';

$search-button-width: 46px;

.topBar {
  width: 100%;
  height: 54px;
  background-color: $ml-yellow;
  border-bottom: 1px solid rgba(51, 51, 51, 0.1);
}

.container {
  height: 54px;

  @media (min-width: 768px) {
    max-width: $body-width;
    margin: 0 auto;
  }

  .logo {
    display: table-cell;
    width: 58px;
    height: 40px;
    padding: 7px 5px;

    & > img {
      width: 58px;
      height: 40px;
    }
  }

  .search {
    display: table-cell;
    vertical-align: top;
    padding: 7px 7px 7px 5px;
    width: 100%;
    height: 54px;
    position: relative;
    box-sizing: border-box;

    & > form {
      display: table;
      width: 100%;

      & > input {
        display: table-cell;
        box-sizing: border-box;
        border: none;
        padding: 5px 30px 5px 15px;
        border-radius: 3px 0 0 3px;
        width: calc(100% - #{$search-button-width});
        height: 40px;
        vertical-align: middle;
        font-size: 18px;

        @media (min-width: 768px) {
          padding: 5px 60px 5px 15px;
        }
      }

      & > button {
        display: table-cell;
        background: url('/assets/magnifiying_glass.png');
        background-size: 18px 18px;
        background-position: center;
        background-repeat: no-repeat;
        width: $search-button-width;
        height: 40px;
        border: none;
        background-color: $ml-lighter-grey;
        border-radius: 0 3px 3px 0;
        vertical-align: middle;
      }
    }
  }
}

