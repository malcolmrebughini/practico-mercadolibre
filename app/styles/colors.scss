$ml-yellow: #FFE600;
$ml-black: #333;
$ml-grey: #666;
$ml-light-grey: #999;
$ml-lighter-grey: #EEE;
