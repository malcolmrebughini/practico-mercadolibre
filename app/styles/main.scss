@import './colors';

body {
  font-family: "Roboto", Helvetica, Arial, sans-serif;
  color: $ml-black;
  background-color: $ml-lighter-grey;
}
