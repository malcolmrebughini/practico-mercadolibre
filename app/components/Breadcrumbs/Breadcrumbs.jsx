import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.scss';


class Breadcrumbs extends React.PureComponent {
  render = () => (
    <div className={styles.container}>
      { this.props.list.map((value, index, array) => (
        <span className={styles.breadCrumb} key={`breadCrumb-${value}`}>
          { array.length !== index + 1 ? `${value} > ` : value }
        </span>
      ))
      }
    </div>
  )
}


Breadcrumbs.propTypes = {
  list: PropTypes.array,
};


export default Breadcrumbs;
