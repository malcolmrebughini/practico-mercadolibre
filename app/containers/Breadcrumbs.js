import { connect } from 'react-redux';
import Breadcrumbs from '../components/Breadcrumbs';


function mapStateToProps(state) {
  return {
    list: state.categories,
  };
}

export default connect(mapStateToProps)(Breadcrumbs);
