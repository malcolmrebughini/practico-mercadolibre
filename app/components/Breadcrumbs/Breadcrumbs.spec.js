import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Breadcrumbs from './Breadcrumbs';
import styles from './styles.scss';

const testKeywordsList = ['this', 'are', 'some', 'breadcrumbs', 'mydude'];

describe('Breadcrumbs', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Breadcrumbs list={testKeywordsList} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render a list of strings as breadcrumbs', () => {

    const wrapper = shallow(
      <Breadcrumbs list={testKeywordsList} />
    );

    const breadcrumbs = wrapper.find(`.${styles.breadCrumb}`);

    expect(breadcrumbs.length).toEqual(5);

    breadcrumbs.forEach((breadcrumb, index) => {
      expect(breadcrumb.text()).toContain(testKeywordsList[index]);
    });
  });
});