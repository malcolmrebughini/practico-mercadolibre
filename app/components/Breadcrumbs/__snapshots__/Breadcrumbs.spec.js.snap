// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render correctly 1`] = `
<div
  className="container"
>
  <span
    className="breadCrumb"
  >
    this &gt; 
  </span>
  <span
    className="breadCrumb"
  >
    are &gt; 
  </span>
  <span
    className="breadCrumb"
  >
    some &gt; 
  </span>
  <span
    className="breadCrumb"
  >
    breadcrumbs &gt; 
  </span>
  <span
    className="breadCrumb"
  >
    mydude
  </span>
</div>
`;
